.claude/CLAUDE.md

```diff
@@ -1,61 +1,122 @@
+  ownership
+- **bitwarden-error**: Error handling across platforms (basic/flat/full modes via proc macro)
+- **bitwarden-encoding**, **bitwarden-uuid**: Encoding and UUID utilities
 
-## Issues necessitating comments
+### 2. Core Infrastructure
 
-**Auto-generated code changes**
+- **bitwarden-core**: Base Client struct extended by feature crates via extension traits. **DO NOT
+  add functionality here - use feature crates instead.**
 
-- Changes to `bitwarden-api-api/` or `bitwarden-api-identity/` are generally discouraged. These are
-  auto-generated from swagger specs.
+### 3. Feature Implementations
 
-**Secrets in logs/errors**
+- **bitwarden-pm**: PasswordManagerClient wrapping core Client, exposes sub-clients: `auth()`,
+  `vault()`, `crypto()`, `sends()`, `generators()`, `exporters()`
+  - Lifecycle: Init → Lock/Unlock → Logout (drops instance)
+  - Storage: Apps use `HashMap<UserId, PasswordManagerClient>`
+- **bitwarden-vault**: Vault item models (encrypted/decrypted states) with Encryptable trait
+- **bitwarden-collections**: Collection models with tree structure via TreeItem trait
+- **bitwarden-auth**: Authentication (send access tokens)
+- **bitwarden-send**: Encrypted temporary secret sharing
+- **bitwarden-generators**: Password/passphrase generators
+- **bitwarden-ssh**: SSH key generation/import
+- **bitwarden-exporters**: Vault export/import with multiple formats
+- **bitwarden-fido**: FIDO2 two-factor authentication
 
-- Do not log keys, passwords, or vault data in logs or error paths. Redact sensitive data.
+### 4. Cross-Platform Bindings
 
-**Business logic in WASM**
+- **bitwarden-api-api**, **bitwarden-api-identity**: Auto-generated API clients (**DO NOT edit -
+  regenerate from OpenAPI specs**)
+- **bitwarden-uniffi**: Mobile bindings (Swift/Kotlin) via UniFFI
+  - Structs: `#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]`
+  - Enums: `#[cfg_attr(feature = "uniffi", derive(uniffi::Enum))]`
+  - Include `uniffi::setup_scaffolding!()` in lib.rs
+- **bitwarden-wasm-internal**: WebAssembly bindings (**thin bindings only - no business logic**)
+  - Structs: `#[derive(Serialize, Deserialize)]` with `#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]`
 
-- `bitwarden-wasm-internal` contains only thin bindings. Move business logic to feature crates.
+## Critical Patterns & Rules
 
-**Unsafe without justification**
+### Cryptography (bitwarden-crypto)
 
-- Any `unsafe` block needs a comment explaining why it's safe and what invariants are being upheld.
+- **DO NOT modify** without careful consideration - backward compatibility is critical
+- **KeyStoreContext**: Never hold across await points (enforced by clippy lint
+  `await_holding_invalid_type`)
+- Naming: `derive_` for deterministic key derivation, `make_` for non-deterministic generation
+- Use `bitwarden_crypto::safe` module first (password-protected key envelope, data envelope)
+  envelope)
```

Comment:
```suggestion
- Use `bitwarden_crypto::safe` module first (password-protected key envelope, data envelope)
  envelope) instead of more low-level primitives
```